pkg/controller/elasticsearch/sset/pod.go

```diff
@@ -20,6 +20,18 @@ import (
 	"github.com/elastic/cloud-on-k8s/pkg/utils/stringsutil"
 )
 
+// HeadlessServiceName returns the name of the headless service for the given StatefulSet.
+func HeadlessServiceName(ssetName string) string {
+	// just use the sset name
+	return ssetName
+}
+
+//PodDNSName returns the DNS resolvable name of the given pod resolvable within the namespace.
+func PodDNSName(pod corev1.Pod) string {
+	ssetName := pod.Labels[label.StatefulSetNameLabelName]
+	return fmt.Sprintf("%s.%s", pod.Name, HeadlessServiceName(ssetName))
```

Comment:
I'm wondering if we should consider how a user may override the Pod hostname and subdomain fields: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields
But I think that's really a corner case that is probably safe to ignore for now. It probably has other impacts.